src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import duke.exceptions.*;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+public class Parser {
+    public Parser(){}
+
+    public static String[] parse(String input, TaskList tasks) {
```

Comment:
Do you think it would be more convenient for you as a developer and more robust for your programe to extract each token, separated by " " (space), first, and then process each token to decide whether it is a command, task name, integer or time? Something like that... 
I see what you are trying to do here. It is just that every time you want to add a new command, you have to modify this "if else" statement and check the length of the input and other things all over again.